db/migrate/20190903030453_add_location_to_microcosms.rb

```diff
@@ -1,13 +1,16 @@
 class AddLocationToMicrocosms < ActiveRecord::Migration[5.2]
   def change
-    change_table "microcosms", :bulk => true do |t|
-      t.string "location", :null => false
-      t.decimal "lat", :null => false
-      t.decimal "lon", :null => false
-      t.integer "min_lat", :null => false
-      t.integer "max_lat", :null => false
-      t.integer "min_lon", :null => false
-      t.integer "max_lon", :null => false
+    # This group of migrations for microcosms will be run together.
+    safety_assured do
```

Comment:
`safety_assured` doesn't work like this! It's a warning that this migration could lock the table (potentially for hours, if there's lots of data) and so you need to find a different approach.

On a production database, this could be a multi-step process of creating shadow columns, backfilling data and so on. See https://github.com/ankane/strong_migrations for more details. However, since this is going to be an empty table, that would be overkill. 

So the best approach for pull requests is to avoid including multiple database migrations. Sure, during development you might realise that the initial database structure was incorrect, and add a fixup migration like this one. But for the PR, please make one migration with your preferred structure. For example, see #1576, where even though there were substantial changes to the database during the development of the PR, there was in the end only one migration.

Reply:
I'll collect migrations later.  I created a ticket to remind me.  https://github.com/openbrian/osm-microcosms/issues/3

Reply:
Rake bitched at me because I was adding columns and setting the default to null at the same time.  Of course that's not going to work on a table that's not empty.  I commented that safety was assured because this migration and the one to create the table would be run together.  When collapsing migrations this goes away.